File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.java
import java.util.*;
import java.io.*;

class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
	public static void main(String[] args) throws IOException
	{
		String str = br.readLine();
		String bomb = br.readLine();
		
		Stack<Character> stack = new Stack<>();
		for(int i = 0; i < str.length(); i++) {
			stack.push(str.charAt(i));
			if(stack.size() < bomb.length()) continue;
			boolean explode = true;
			for(int check = 0; check < bomb.length(); check++) {
				if(stack.get(stack.size()-1-check) != bomb.charAt(bomb.length()-1-check)) {
					explode = false;
					break;
				}
			}
			if(explode) {
				for(int j=0; j<bomb.length(); j++) {
					stack.pop();
				}
			}
		}
		if(stack.size() > 0) {
			for(char c : stack) {
				bw.append(Character.toString(c));
			}
		}else {
			bw.append("FRULA");
		}
		
		bw.flush();
	}
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
def explode(string, bomb):
    bomb_size = len(bomb)
    if len(string) < bomb_size:
        return string
    stack = []
    for i in range(len(string)):
        stack.append(string[i])
        if stack[-1] == bomb[-1]:
            if "".join(stack[-bomb_size:]) == bomb:
                del stack[-bomb_size:]
    if stack:
        return "".join(stack)
    return "FRULA"
            
# 데이터 입력
sentence = input()
bomb = input()
print(explode(sentence, bomb))